crates/bevy_sprite/src/render/mod.rs

```diff
@@ -480,11 +480,15 @@ pub fn queue_sprites(
                 // By default, the size of the quad is the size of the texture
                 let mut quad_size = current_image_size;
 
-                // If a rect is specified, adjust UVs and the size of the quad
+                // If a rect is specified, adjust UVs and the size of the quad.
+                // The UVs are shrunk by half the size of a texel to prevent
+                // floating point inaccuracies from causing incorrect sampling.
                 if let Some(rect) = extracted_sprite.rect {
                     let rect_size = rect.size();
                     for uv in &mut uvs {
```

Comment:
```suggestion
                    for &uv in &mut uvs {
```

Does the borrow-checker let us get away with doing this? It would be really nice to avoid the derefs everywhere for legibility reasons, particularly with the multiplications.

Reply:
I don't think so - without a reference the value would either be a copy or moved and neither of those works.